[2024-09-15T20:06:34.639+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-15T20:06:34.654+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [queued]>
[2024-09-15T20:06:34.662+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [queued]>
[2024-09-15T20:06:34.662+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-09-15T20:06:34.677+0000] {taskinstance.py:2888} INFO - Executing <Task(operatorSaveFile): SaveFile> on 2024-09-14 00:00:00+00:00
[2024-09-15T20:06:34.682+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=99) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-09-15T20:06:34.683+0000] {standard_task_runner.py:72} INFO - Started process 100 to run task
[2024-09-15T20:06:34.684+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'SitesWebScraping', 'SaveFile', 'scheduled__2024-09-14T00:00:00+00:00', '--job-id', '256', '--raw', '--subdir', 'DAGS_FOLDER/DagWebScraping.py', '--cfg-path', '/tmp/tmpqdvzekga']
[2024-09-15T20:06:34.686+0000] {standard_task_runner.py:105} INFO - Job 256: Subtask SaveFile
[2024-09-15T20:06:34.729+0000] {task_command.py:467} INFO - Running <TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [running]> on host db31d856c8fa
[2024-09-15T20:06:34.800+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='SitesWebScraping' AIRFLOW_CTX_TASK_ID='SaveFile' AIRFLOW_CTX_EXECUTION_DATE='2024-09-14T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-09-14T00:00:00+00:00'
[2024-09-15T20:06:34.802+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-15T20:06:34.823+0000] {logging_mixin.py:190} INFO - /opt/***/datalake/rawzone/kabum
[2024-09-15T20:06:34.825+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/operators/OperatorSaveFile.py", line 20, in execute
    with open(filepath, 'w') as aquivo:
         ^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/opt/airflow/datalake/rawzone/kabum/data'
[2024-09-15T20:06:34.835+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=SitesWebScraping, task_id=SaveFile, run_id=scheduled__2024-09-14T00:00:00+00:00, execution_date=20240914T000000, start_date=20240915T200634, end_date=20240915T200634
[2024-09-15T20:06:34.846+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-15T20:06:34.847+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 256 for task SaveFile ([Errno 2] No such file or directory: '/opt/airflow/datalake/rawzone/kabum/data'; 100)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/operators/OperatorSaveFile.py", line 20, in execute
    with open(filepath, 'w') as aquivo:
         ^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/opt/airflow/datalake/rawzone/kabum/data'
[2024-09-15T20:06:34.858+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-09-15T20:06:34.872+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-15T20:06:34.876+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-09-15T20:19:57.599+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-15T20:19:57.616+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [queued]>
[2024-09-15T20:19:57.624+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [queued]>
[2024-09-15T20:19:57.625+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-09-15T20:19:57.640+0000] {taskinstance.py:2888} INFO - Executing <Task(operatorSaveFile): SaveFile> on 2024-09-14 00:00:00+00:00
[2024-09-15T20:19:57.647+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=100) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-09-15T20:19:57.648+0000] {standard_task_runner.py:72} INFO - Started process 101 to run task
[2024-09-15T20:19:57.649+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'SitesWebScraping', 'SaveFile', 'scheduled__2024-09-14T00:00:00+00:00', '--job-id', '264', '--raw', '--subdir', 'DAGS_FOLDER/DagWebScraping.py', '--cfg-path', '/tmp/tmpf0udyu0h']
[2024-09-15T20:19:57.651+0000] {standard_task_runner.py:105} INFO - Job 264: Subtask SaveFile
[2024-09-15T20:19:57.701+0000] {task_command.py:467} INFO - Running <TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [running]> on host f2f338af217f
[2024-09-15T20:19:57.779+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='SitesWebScraping' AIRFLOW_CTX_TASK_ID='SaveFile' AIRFLOW_CTX_EXECUTION_DATE='2024-09-14T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-09-14T00:00:00+00:00'
[2024-09-15T20:19:57.780+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-15T20:19:57.806+0000] {logging_mixin.py:190} INFO - /opt/***/datalake/rawzone/kabum=2024-09-15
[2024-09-15T20:19:57.822+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-15T20:19:57.822+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=SitesWebScraping, task_id=SaveFile, run_id=scheduled__2024-09-14T00:00:00+00:00, execution_date=20240914T000000, start_date=20240915T201957, end_date=20240915T201957
[2024-09-15T20:19:57.863+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-09-15T20:19:57.879+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-15T20:19:57.882+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-09-15T20:37:45.622+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-15T20:37:45.652+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [queued]>
[2024-09-15T20:37:45.684+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [queued]>
[2024-09-15T20:37:45.685+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-09-15T20:37:45.703+0000] {taskinstance.py:2888} INFO - Executing <Task(operatorSaveFile): SaveFile> on 2024-09-14 00:00:00+00:00
[2024-09-15T20:37:45.709+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=126) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-09-15T20:37:45.710+0000] {standard_task_runner.py:72} INFO - Started process 127 to run task
[2024-09-15T20:37:45.710+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'SitesWebScraping', 'SaveFile', 'scheduled__2024-09-14T00:00:00+00:00', '--job-id', '272', '--raw', '--subdir', 'DAGS_FOLDER/DagWebScraping.py', '--cfg-path', '/tmp/tmpqstikm5a']
[2024-09-15T20:37:45.713+0000] {standard_task_runner.py:105} INFO - Job 272: Subtask SaveFile
[2024-09-15T20:37:45.775+0000] {task_command.py:467} INFO - Running <TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [running]> on host 6370f38be46f
[2024-09-15T20:37:45.863+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='SitesWebScraping' AIRFLOW_CTX_TASK_ID='SaveFile' AIRFLOW_CTX_EXECUTION_DATE='2024-09-14T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-09-14T00:00:00+00:00'
[2024-09-15T20:37:45.864+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-15T20:37:45.911+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-15T20:37:45.912+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=SitesWebScraping, task_id=SaveFile, run_id=scheduled__2024-09-14T00:00:00+00:00, execution_date=20240914T000000, start_date=20240915T203745, end_date=20240915T203745
[2024-09-15T20:37:45.965+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-09-15T20:37:45.985+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-15T20:37:45.991+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-09-15T20:40:10.819+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-15T20:40:10.839+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [queued]>
[2024-09-15T20:40:10.848+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [queued]>
[2024-09-15T20:40:10.849+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-09-15T20:40:10.873+0000] {taskinstance.py:2888} INFO - Executing <Task(operatorSaveFile): SaveFile> on 2024-09-14 00:00:00+00:00
[2024-09-15T20:40:10.878+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=172) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-09-15T20:40:10.881+0000] {standard_task_runner.py:72} INFO - Started process 176 to run task
[2024-09-15T20:40:10.882+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'SitesWebScraping', 'SaveFile', 'scheduled__2024-09-14T00:00:00+00:00', '--job-id', '276', '--raw', '--subdir', 'DAGS_FOLDER/DagWebScraping.py', '--cfg-path', '/tmp/tmpfsjrftqi']
[2024-09-15T20:40:10.886+0000] {standard_task_runner.py:105} INFO - Job 276: Subtask SaveFile
[2024-09-15T20:40:10.969+0000] {task_command.py:467} INFO - Running <TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [running]> on host 6370f38be46f
[2024-09-15T20:40:11.077+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='SitesWebScraping' AIRFLOW_CTX_TASK_ID='SaveFile' AIRFLOW_CTX_EXECUTION_DATE='2024-09-14T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-09-14T00:00:00+00:00'
[2024-09-15T20:40:11.078+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-15T20:40:11.129+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-15T20:40:11.131+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=SitesWebScraping, task_id=SaveFile, run_id=scheduled__2024-09-14T00:00:00+00:00, execution_date=20240914T000000, start_date=20240915T204010, end_date=20240915T204011
[2024-09-15T20:40:11.339+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-09-15T20:40:11.356+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-15T20:40:11.545+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-09-15T20:49:58.080+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-15T20:49:58.100+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [queued]>
[2024-09-15T20:49:58.110+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [queued]>
[2024-09-15T20:49:58.111+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-09-15T20:49:58.130+0000] {taskinstance.py:2888} INFO - Executing <Task(operatorSaveFile): SaveFile> on 2024-09-14 00:00:00+00:00
[2024-09-15T20:49:58.135+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=120) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-09-15T20:49:58.137+0000] {standard_task_runner.py:72} INFO - Started process 121 to run task
[2024-09-15T20:49:58.137+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'SitesWebScraping', 'SaveFile', 'scheduled__2024-09-14T00:00:00+00:00', '--job-id', '285', '--raw', '--subdir', 'DAGS_FOLDER/DagWebScraping.py', '--cfg-path', '/tmp/tmp84qhst0a']
[2024-09-15T20:49:58.139+0000] {standard_task_runner.py:105} INFO - Job 285: Subtask SaveFile
[2024-09-15T20:49:58.184+0000] {task_command.py:467} INFO - Running <TaskInstance: SitesWebScraping.SaveFile scheduled__2024-09-14T00:00:00+00:00 [running]> on host 6370f38be46f
[2024-09-15T20:49:58.262+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='SitesWebScraping' AIRFLOW_CTX_TASK_ID='SaveFile' AIRFLOW_CTX_EXECUTION_DATE='2024-09-14T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-09-14T00:00:00+00:00'
[2024-09-15T20:49:58.264+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-15T20:49:58.325+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/operators/OperatorSaveFile.py", line 18, in execute
    json_str=json.dumps(self.data)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type set is not JSON serializable
[2024-09-15T20:49:58.342+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=SitesWebScraping, task_id=SaveFile, run_id=scheduled__2024-09-14T00:00:00+00:00, execution_date=20240914T000000, start_date=20240915T204958, end_date=20240915T204958
[2024-09-15T20:49:58.381+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-15T20:49:58.382+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 285 for task SaveFile (Object of type set is not JSON serializable; 121)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/operators/OperatorSaveFile.py", line 18, in execute
    json_str=json.dumps(self.data)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type set is not JSON serializable
[2024-09-15T20:49:58.432+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-09-15T20:49:58.449+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-15T20:49:58.452+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
